### PR TITLE
MAP-19 Replace global properties with metadata mappings

### PR DESCRIPTION
**Pull request by PawelGutkowski**
This PR depends on https://github.com/openmrs/openmrs-module-metadatamapping/pull/34

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/emrapi/EmrApiActivator.java [hunk 1]
```diff
@@ -94,20 +119,183 @@ private void ensurePrivilegeLevelRoles() {
...
+        }.saveIfMissing(EmrApiConstants.GP_UNKNOWN_LOCATION);
+
+        new MetadataMappingHelper<PatientIdentifierType>(emrapiMetadataSource) {
+            @Override
+            PatientIdentifierType getMetadataByUuid(String uuid) {
+                return Context.getPatientService().getPatientIdentifierTypeByUuid(uuid);
+            }
+        }.saveIfMissing(EmrApiConstants.PRIMARY_IDENTIFIER_TYPE);
+
+        new MetadataMappingHelper<VisitType>(emrapiMetadataSource) {
```
Comment (rkorytkowski) [inline api/src/main/java/org/openmrs/module/emrapi/EmrApiActivator.java]: Rename to GlobalPropertyToMappingConverter

File: api/src/main/java/org/openmrs/module/emrapi/EmrApiActivator.java [hunk 2]
```diff
@@ -94,20 +119,183 @@ private void ensurePrivilegeLevelRoles() {
...
+            @Override
+            VisitType getMetadataByUuid(String uuid) {
+                return Context.getVisitService().getVisitTypeByUuid(uuid);
+            }
+        }.saveIfMissing(EmrApiConstants.GP_AT_FACILITY_VISIT_TYPE);
```
Comment (rkorytkowski) [inline api/src/main/java/org/openmrs/module/emrapi/EmrApiActivator.java]: Rename to convert(String globalProperty)

**Comment by rkorytkowski** [comment]
**Not for merging before Reference Application 2.5 release.**

**Comment by dkayiwa** [comment]
@PawelGutkowski now that we have released 2.5, do you think its time to resolve these conflicts and we merge? :)

**Comment by PawelGutkowski** [comment]
@dkayiwa it depends on if your, @rkorytkowski and other devs involved in EMR-API opinion. If you think it is ready, I will resolve conflicts so we can merge it and dependinig PRs :)

**Comment by dkayiwa** [comment]
@PawelGutkowski yes go ahead and resolve the conflicts so that we can merge. :)

**Comment by PawelGutkowski** [comment]
conflicts resolved :)

**Comment by dkayiwa** [comment]
@PawelGutkowski the code looks good. I have tried to actually test it out before merging by doing a fresh installation of the reference application but ended up with this error: java.lang.IllegalStateException: Configuration required: emr.primaryIdentifierType

Did you try it out on a fresh install?

**Comment by dkayiwa** [comment]
@PawelGutkowski ignore the above comment. I realized that i needed to also compile the referenceapplication and referencemetadata modules with the changes in the followup pull requests that you made. :)

**Pull request merged**